crates/extensions/c8y_mapper_ext/src/operations/handlers/device_profile.rs

```diff
@@ -1811,4 +1813,96 @@ mod tests {
         // Expect `506` smartrest message on `c8y/s/us`.
         assert_received_contains_str(&mut mqtt, [("c8y/s/us", "506,123456")]).await;
     }
+
+    #[tokio::test]
+    async fn skip_device_profile_operation() {
+        let ttd = TempTedgeDir::new();
+        let test_handle = spawn_c8y_mapper_actor(&ttd, true).await;
+        let TestHandle { mqtt, .. } = test_handle;
+        let mut mqtt = mqtt.with_timeout(TEST_TIMEOUT_MS);
+
+        skip_init_messages(&mut mqtt).await;
+
+        // Simulate config_update command with "successful" state
+        mqtt.send(MqttMessage::new(
+            &Topic::new_unchecked("te/device/main///cmd/device_profile/c8y-mapper-123456"),
+            json!({
+                "status": "successful",
+                "name": "test-profile",
+                "operations": [
+                    {
+                        "operation": "firmware_update",
+                        "@skip": false,
+                        "payload": {
+                            "name": "test-firmware",
+                            "version": "1.0",
+                            "remoteUrl": "http://www.my.url"
+                        }
+                    },
+                    {
+                        "operation": "software_update",
+                        "@skip": false,
+                        "payload": {
+                            "updateList": [
+                                {
+                                    "type": "apt",
+                                    "modules": [
+                                        {
+                                            "name": "test-software-1",
+                                            "version": "latest",
+                                            "action": "install"
+                                        },
+                                        {
+                                            "name": "test-software-2",
+                                            "version": "latest",
+                                            "action": "install"
+                                        }
+                                    ]
+                                }
+                            ]
+                        }
+                    },
+                    {
+                        "operation": "config_update",
+                        "@skip": true,
```

Comment:
I don't get how this test checks that the config update is skipped, but software & firmware update applied.